src/coreclr/jit/codegenarm64.cpp

```diff
@@ -3846,17 +3846,28 @@ void CodeGen::genEmitHelperCall(unsigned helper, int argSize, emitAttr retSize,
 void CodeGen::genSIMDIntrinsic(GenTreeSIMD* simdNode)
 {
     // NYI for unsupported base types
-    if (simdNode->GetSimdBaseType() != TYP_INT && simdNode->GetSimdBaseType() != TYP_LONG &&
-        simdNode->GetSimdBaseType() != TYP_FLOAT && simdNode->GetSimdBaseType() != TYP_DOUBLE &&
-        simdNode->GetSimdBaseType() != TYP_USHORT && simdNode->GetSimdBaseType() != TYP_UBYTE &&
-        simdNode->GetSimdBaseType() != TYP_SHORT && simdNode->GetSimdBaseType() != TYP_BYTE &&
-        simdNode->GetSimdBaseType() != TYP_UINT && simdNode->GetSimdBaseType() != TYP_ULONG)
-    {
-        // We don't need a base type for the Upper Save & Restore intrinsics, and we may find
-        // these implemented over lclVars created by CSE without full handle information (and
-        // therefore potentially without a base type).
-        if ((simdNode->gtSIMDIntrinsicID != SIMDIntrinsicUpperSave) &&
+    if (!TypeIsInList(simdNode->GetSimdBaseType(), TYP_INT, TYP_LONG, TYP_FLOAT, TYP_DOUBLE, TYP_USHORT, TYP_UBYTE,
```

Comment:
This check is probably better served by one that does a `value >= TYP_BYTE && value <= TYP_ULONG` (I think we have an existing helper for this already), since these values are sequentially ordered in the enum.

Reply:
I believe the relevant helper is `varTypeIsArithmetic`, although it also allows `TYP_BOOL`.

Reply:
`varTypeIsArithmetic` is the one that the rest of the SIMD checks use.

Reply:
Do we have tests for TYP_BOOL SIMD intrinsic? why don't they fail nowadays on this check?

Reply:
Because we filter it out by resolving the actual handle for all generic types during import: <https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/simd.cpp#L128>

If the `CorInfoType` isn't recognized here, then we return `CORINFO_TYPE_UNDEF` and that is ultimately filtered out and treated as "unsupported":
* <https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/simd.cpp#L1922-L1925>
* <https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/simdashwintrinsic.cpp#L207-L213>
* <https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/hwintrinsic.cpp#L662-L729>
* etc

Reply:
And yes, we have tests (such as https://github.com/dotnet/runtime/tree/main/src/tests/JIT/HardwareIntrinsics/General/NotSupported) that covers types like `bool` and `nint`/`nuint` (the latter two of which are only supported on `Vector<T>` and not yet on `Vector64/128/256<T>`)